lib/cext/preprocess.rb

```diff
@@ -39,6 +39,41 @@
 }
 EOF
 
+MYSQL2_FREE_WRAPPER = <<-EOF
+    nogvl_close(wrapper);
+    truffle_release_handle(wrapper->encoding);
+    truffle_release_handle(wrapper->active_thread);
+EOF
+
+MYSQL2_FREE_RESULT_WRAPPER = <<-EOF
+  truffle_release_handle(wrapper->fields);
+  truffle_release_handle(wrapper->rows);
+  truffle_release_handle(wrapper->client);
+  truffle_release_handle(wrapper->encoding);
+  truffle_release_handle(wrapper->statement);
+  xfree(wrapper);
+EOF
+
+MYSQL2_FREE_STATEMENT = <<-EOF
+if (stmt_wrapper->refcount == 0) {
+  rb_tr_release_handle(stmt_wrapper->client);
+EOF
+
+def read_field(struct_var_name, field_name)
+  { 
+    match: /\b#{struct_var_name}(\.|->)#{field_name}\s*([\),;]|==|!=)/,
+    replacement: "rb_tr_managed_from_handle(#{struct_var_name}\\1#{field_name}) \\2"
+  }
+end
+
+def write_field(struct_var_name, field_name, leaking)
+  leaking_str = leaking ? '_leaking': ''
+  {
+    match: /\b#{struct_var_name}(\.|->)#{field_name}\s*=\s*(\w.+);\s*$/ ,
+    replacement: "#{struct_var_name}\\1#{field_name} = rb_tr_handle_for_managed#{leaking_str}(\\2);"
```

Comment:
I would also capture around the `\s*=\s*` here to preserve the original spacing.

Reply:
Fixed.